qopt_best_practices/swap_strategies/build_circuit.py

```diff
@@ -144,39 +152,70 @@ def create_qaoa_swap_circuit(
             its output must have the same ordering of qubits as its input.
     """
 
+    # Save the parameters of the original, total Hamiltonian 
     num_qubits = cost_operator.num_qubits
-
-    if theta is not None:
-        gamma = theta[: len(theta) // 2]
-        beta = theta[len(theta) // 2:]
-        qaoa_layers = len(theta) // 2
-    else:
-        gamma = beta = None
-        qaoa_layers = qaoa_layers
-
-    # First, create the ansatz of 1 layer of QAOA without mixer
-    cost_layer = QAOAAnsatz(
-        cost_operator,
+    gate_list = cost_operator.paulis
+    weights_list = cost_operator.coeffs
+    
+    # Prepare the separate two parts of the total Hamiltonian
+    cost_operator_order1_only = QuantumCircuit(num_qubits)
+    cost_operator_order2_only = []
+
+    # Create H2 as an operator
+    for pauli, gate_weight in zip(gate_list, weights_list):
+        if sum(pauli.x) != 0 or sum(pauli.z) > 2:
+            raise Exception("This method can only handle first order and second order Pauli Z terms.")
```

Comment:
```suggestion
            raise NotImplementedError("This method can only handle first order and second order Pauli Z terms.")
```
I think it would be good to later on handle higher-order terms and then use the standard Qiskit transpiler to start of with.